external/Apr1Lib.cmake

```diff
@@ -51,24 +51,47 @@ if (UNIX)
         set(aprlib_config_options ${aprlib_config_options} --enable-debug)
     endif()
 
-    ExternalProject_Add(Apr1StaticLib
-        URL ${aprlib_url}
-
-        UPDATE_COMMAND ""
-        PATCH_COMMAND ""
-
-        CONFIGURE_COMMAND
-            <SOURCE_DIR>/configure
-                --prefix=${aprlib_install_prefix}
-                ${aprlib_config_options}
-                CFLAGS=${aprlib_cflags}
-
-        BUILD_COMMAND
-            make -f Makefile all
-
-        INSTALL_COMMAND
-            make -f Makefile install
-    )
+    # gcc v4.9 requires its own binutils-wrappers for LTO (flag -flto)
+    # fixes #981
+    if(CMAKE_COMPILER_IS_GNUCXX AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "4.9" OR CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "4.9"))
+        ExternalProject_Add(Apr1StaticLib
+        	URL ${aprlib_url}
+
+        	UPDATE_COMMAND ""
+        	PATCH_COMMAND ""
+
+        	CONFIGURE_COMMAND
+            	<SOURCE_DIR>/configure AR=gcc-ar NM=gcc-nm RANLIB=gcc-ranlib
+                	--prefix=${aprlib_install_prefix}
+                	${aprlib_config_options}
+                	CFLAGS=${aprlib_cflags}
+
+        	BUILD_COMMAND
+            	make -f Makefile all
+
+        	INSTALL_COMMAND
+            	make -f Makefile install
+    	)
+    else()
```

Comment:
Yeah, you're right. It would be best if you could use an if-statement within the ExternalProject_Add() function block, but that doesn't work (cmake just ignores the if-condition and silently executes the contents of the if-block no matter what the condition is).